scripts/usb_scan.py

```diff
@@ -0,0 +1,660 @@
+#!/usr/bin/env python
+#
+# Copyright (C) Citrix Systems Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation; version 2.1 only. #
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# package dependency: python-pyudev
+#
+# called by XAPI to do
+# 1. a "lsusb" operation: get all usb device and interface information
+# 2. check if device can be passed through based on policy file
+# 3. return the device info to XAPI in json format
+
+import abc
+import argparse
+import json
+import xcp.logger as log
+import logging
+import pyudev
+import re
+import traceback
+
+# True to print log to user.log
+__DIAGNOSTIC = False
```

Comment:
Can you avoid setting and checking `_DIAGNOSTIC` and instead do something like the following in the main function?

```python
if args.diagnostic:
    log.logToSyslog(level=logging.DEBUG)
else:
    log.logToSyslog(level=logging.WARNING)
```

Reply:
Those are two ways to enable debug mode: change the file and change the parameter.

Reply:
Maybe I am missing something, otherwise I don't like the need of a special wrapper to change the log behaviour when changing the loglevel is equivalent.

Reply:
sure. done.